app/models/concerns/stock_settings_override_validation.rb

```diff
@@ -46,7 +46,9 @@ def disallow_count_on_hand_if_on_demand
   end
 
   def require_count_on_hand_if_limited_stock
-    return unless on_demand == false && count_on_hand.blank?
```

Comment:
I think this one is correct for this specific error, we're returning if `on_demand` is `false` and `count_on_hand` is `nil`. That is; if `on_demand` is `true`, `count_on_hand` can't be `nil`. It's correct (although I realise this is a bit crazy).

I think what we need is a new method to validate: if `count_on_hand` is _present_ (not nil), it has to be numeric :+1:

Reply:
Or... we could probably just define it in the `VariantOverride` model along with price, and use:
```
validates :count_on_hand, numericality: { greater_than_or_equal_to: 0 }, allow_nil: true
```

Reply:
that shound like a better aproach! Will do that